src/integrations/buckaroo/class-buckaroo.php

```diff
@@ -0,0 +1,571 @@
+   */
+  public function set_test(): void {
+    $this->is_test_uri = true;
+  }
+
+  /**
+   * Set if you need to use the test URI instead of live one.
+   *
+   * @return void
+   */
+  public function set_data_request(): void {
+    $this->is_data_request = true;
+  }
+
+  /**
+   * Set's currency as uppercase 3-letter string (example: EUR)
+   *
+   * @param  string $currency Currency string.
+   * @return void
+   */
+  public function set_currency( string $currency ): void {
+    $this->currency = $currency;
+  }
+
+  /**
+   * Getter for $this->currency.
+   *
+   * @return string
+   */
+  public function get_currency(): string {
+    return $this->currency;
+  }
+
+  /**
+   * Get type of payment. Defaults to iDEAL.
+   *
+   * @return  string
+   */
+  public function get_pay_type() {
+     return $this->pay_type;
+  }
+
+  /**
+   * Set type of payment. Defaults to iDEAL.
+   *
+   * @param  string $pay_type  Type of payment. Defaults to iDEAL.
+   *
+   * @return  self
+   */
+  public function set_pay_type( string $pay_type ) {
+    $this->pay_type = $pay_type;
+    return $this;
+  }
+
+  /**
+   * Get return URL after payment.
+   *
+   * @return  string
+   */
+  public function get_return_url() {
+    return $this->return_url;
+  }
+
+  /**
+   * Set return URL after payment.
+   *
+   * @param  string $return_url  Return URL after payment.
+   *
+   * @return  self
+   */
+  public function set_return_url( string $return_url ) {
+    $this->return_url = $return_url;
+
+    return $this;
+  }
+
+  /**
+   * Get return URL after payment cancel.
+   *
+   * @return  string
+   */
+  public function get_return_url_cancel() {
+     return $this->return_url_cancel;
+  }
+
+  /**
+   * Set return URL after payment cancel.
+   *
+   * @param  string $return_url_cancel  Return URL after payment cancel.
+   * @return  self
+   */
+  public function set_return_url_cancel( string $return_url_cancel ) {
+    $this->return_url_cancel = $return_url_cancel;
+
+    return $this;
+  }
+
+  /**
+   * Get return URL after payment error.
+   *
+   * @return  string
+   */
+  public function get_return_url_error() {
+     return $this->return_url_error;
+  }
+
+  /**
+   * Set return URL after payment error.
+   *
+   * @param  string $return_url_error  Return URL after payment error.
+   *
+   * @return  self
+   */
+  public function set_return_url_error( string $return_url_error ) {
+    $this->return_url_error = $return_url_error;
+
+    return $this;
+  }
+
+  /**
+   * Get return URL after payment reject.
+   *
+   * @return  string
+   */
+  public function get_return_url_reject() {
+     return $this->return_url_reject;
+  }
+
+  /**
+   * Set return URL after payment reject.
+   *
+   * @param  string $return_url_reject  Return URL after payment reject.
+   *
+   * @return  self
+   */
+  public function set_return_url_reject( string $return_url_reject ) {
+    $this->return_url_reject = $return_url_reject;
+
+    return $this;
+  }
```

Comment:
`public` methods should be before `private` and `protected` ones. 

And only use `protected` if you want to extend the class, otherwise use `private`.

Reply:
Yeah I was thinking about this. But does it make sense to have setters / getters on top? I always put them on bottom because, you know, you don't really need to look at them 99% of the time. I dunno, I also agree that we should have order (public methods first, then protected then private), not really sure what is better here.

Reply:
it adds an unnecessary boilerplate, but it's a common practice to have them on top. Plus `public`, `protected`, `private` is also common in most places I've seen (I know at WPCS it's mandatory).

I've seen a package in one of the PHP meetups that kinda extracts getters and setters, cannot remember what was the name :/ But it reduces boilerplate code a lot

Reply:
Ok, sounds good, will move them to top!